### PR TITLE
GEOMESA-1149 Improving error message for schema version mismatches

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala [hunk 1]
```diff
@@ -655,7 +655,18 @@ class AccumuloDataStore(val connector: Connector,
       val sft = SimpleFeatureTypes.createType(name.getURI, attributes)
       // IMPORTANT: set data that we want to pass around with the sft
       metadata.read(featureName, DTGFIELD_KEY).foreach(sft.setDtgField)
-      sft.setSchemaVersion(metadata.readRequired(featureName, VERSION_KEY).toInt)
+      val version = metadata.readRequired(featureName, VERSION_KEY).toInt
+      if (version > CURRENT_SCHEMA_VERSION) {
+        logger.error(s"Trying to access schema ${sft.getTypeName} with version $version" +
```
Comment (jahhulbert-ccri) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala]: might need space after ```$version``` in the string

**Comment by jahhulbert-ccri** [comment]
looks good...seeing if anyone else has comments

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala (hunk 1 above)
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala]: maybe... :)

Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala]: fixed.

**Comment by elahrvivaz** [comment]
build failure seems to be unrelated to this pr... an operation took longer than expected and failed the expectation in geomesa-convert.

**Comment by jahhulbert-ccri** [comment]
Rekicked the build. Gonna merge

**Comment by jahhulbert-ccri** [comment]
be79a09e96f7b7c0c2d998c6ec936303eb9d3233

**Pull request closed without merging**